### PR TITLE
Missions

### PR DESCRIPTION
**Pull request by Chike1990**
In this task, I worked with the real live data from the SpaceX API. I built a web application for a company that provides commercial and scientific space travel services. The application allows users to book rockets and join selected space missions.

### CODE REVIEW

File: src/services/missionsService.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/* eslint-disable import/prefer-default-export */
```
Comment (Dipeshtwis) [inline src/services/missionsService.js]: - [ ] I would not recommend disabling linter, It's better to improve the code.

File: src/services/rocketsService.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
-import axios from "axios";
-import { rocketsUrl } from "./urls";
+/* eslint-disable import/prefer-default-export */
```
Comment (Dipeshtwis) [inline src/services/rocketsService.js]: - [ ] I would not recommend disabling linter, It's better to improve the code.

File: src/views/Missions.css [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  padding-top: 12px;
+  padding-bottom: 12px;
+  text-align: left;
+}
+
+/* tr, td {
+  border: 1px solid #ccc;
+  padding: 10px;
+  display: inline-block;
+} */
```
Comment (Dipeshtwis) [inline src/views/Missions.css]: - [ ] I would recommend removing the commented code to make the codebase cleaner.

File: README.md [hunk 4]
```diff
@@ -1,70 +1,61 @@
...
+```
+<details align="right">
+<summary><small>What are these commands?</summary>
+- the `$ cd` command is used to move to different folders.<br>
+- while `$ npm run build` is used to compile the aplication files.</small>
+</details>
 
-### `yarn build`
+### Live Server
+Optionally, you can run the application [here]() or [here]().
```
Comment (Dipeshtwis) [inline README.md]: - [ ] I would recommend including the live server, or either remove the section to make the README professional.

File: src/views/Profile.jsx [hunk 5]
```diff
@@ -1,7 +1,29 @@
...
   return (
-    <div>
-      Profile Component
+    <div className="flex-row justify-around Profile">
+      <div style={{width: "35%"}}>
+        <h2>My Missions</h2>
+        <div className="Profile__left">
+          {
+            missions.map(({mission_name, reserved}) => reserved && (<p>{ mission_name }</p>))
+          }
```
Comment (Dipeshtwis) [inline src/views/Profile.jsx]: - [ ] I would recommend rendering a list of all joined missions using the `filter()` method as it is the best approach and it will give you all the filter results.

File: src/views/Profile.jsx [hunk 6]
```diff
@@ -1,7 +1,29 @@
...
+        </div>
+      </div>
+      <div style={{width: "35%"}}>
+        <h2>My Rockets</h2>
+        <div className="Profile__right">
+          {
+            rockets.map(({rocket_name, reserved}) => reserved && (<p>{ rocket_name }</p>))
+          }
```
Comment (Dipeshtwis) [inline src/views/Profile.jsx]: - [ ] I would recommend rendering a list of all reserved rockets using the `filter()` method as it is the best approach and it will give you all the filter results.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @Chike1990, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: The app is working fine.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

- [ ] Please write unit tests with React Testing Library and Jest (for unit tests). As test case covers all the edge cases and it will ensure that your app is really working fine.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi **Team**        
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Good use of gitflow :+1: 

## Required Changes ♻️
- [ ] Unfortunately, I am unable to run your project because of some linter errors that it is showing only locally. KIndly see the pictures below:
Screenshot from 2021-10-10 01-27-28
Screenshot from 2021-10-10 01-26-55


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 7]
```diff
@@ -1,70 +1,58 @@
...
+2. After, run the following commands in your terminal:
 
-Launches the test runner in the interactive watch mode.\
-See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.
+```sh
+$ cd Space-Travelers-Hub
+$ npm install
+$ npm run build
+$ cd dist
+```
```
Comment (cyonii) [inline README.md:30]: - [ ] [OPTIONAL] You did a great job providing setup commands, but this command is not quite clear to non-technical users. For example, you did not specify what to be done after the user `cd` into the `dist` directory. Kindly give a clear explanation so that anyone can set up this project. Also, consider adding instructions to run the development server.

**Review by cyonii — CHANGES_REQUESTED** [review]
Hi @Chike1990 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._
- [ ] Kindly write unit test as requested by the previous reviewer
- [ ] I was unable to run the project, it outputted a bunch of errors on the console (see image below). Kindly fix this error and resubmit
Screen Shot 2021-10-10 at 12 44 20 AM
Screen Shot 2021-10-10 at 12 55 00 AM
Screen Shot 2021-10-10 at 12 55 14 AM

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Rocket.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
+/* eslint-disable react/prop-types */
```
Comment (wandji20) [inline src/components/Rocket.js:1]: - [ optional ] It will be nice to install `prop-types`  and define the types for each prop

File: src/tests/rockets.test.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
+// import { reserveRocket, setRockets } from '../redux/rockets/rockets-actions';
```
Comment (wandji20) [inline src/tests/rockets.test.js]: - [ ] It will be nice to remove all comments from your codebase.

File: src/views/Missions.js [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+      .then((response) => {
+        const missions = response.data.map((mission) => ({
+          missionName: mission.mission_name,
+          missionId: mission.mission_id,
+          description: mission.description,
+          reserved: false,
+        }));
+        dispatch(setMissions(missions));
+      })
+      .catch((error) => console.log(error));
```
Comment (wandji20) [inline src/views/Missions.js]: - [ ] It will be nice to remove all `console.log` statements from your codebase.

File: src/views/Profile.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+              .map(({ missionName }) => (<p>{missionName}</p>))
+          }
+        </div>
+      </div>
+      <div style={{ width: '35%' }}>
+        <h2>My Rockets</h2>
+        <div className="Profile__right">
+          {
+            rockets.filter(({ reserved }) => reserved === true)
+              .map(({ rocketName }) => (<p>{rocketName}</p>))
```
Comment (wandji20) [inline src/views/Profile.js:25]: - [ optional ] It is good practice to include keys when rendering a list.

File: src/views/Rockets.js [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+        const rockets = data.map((rocket) => ({
+          id: rocket.id,
+          rocketName: rocket.rocket_name,
+          description: rocket.description,
+          flickrImages: rocket.flickr_images,
+          reserved: false,
+        }));
+        dispatch(setRockets(rockets));
+      })
+      .catch((error) => console.log(error));
```
Comment (wandji20) [inline src/views/Rockets.js]: - [ ] As mentioned already, let's remove all instances of `console.log` statements

**Review by wandji20 — CHANGES_REQUESTED** [review]
Hello, @Chike1990  :raised_hands: 
-----
Great :muscle:  job :raised_hands:  implementing the review changes.
However, there are a few issues you need to address to have this project approved.

## Status: Require Changes :recycle: :red_circle: 
Please check the comments below and under the review and make all required changes.
Optional comments are not required to mark this project approved
- [ ] Some tests are falling. You can fix this.
- [ ] Please endeavor to remove all commented code and `console.log` statements from your codebase.
Screenshot from 2021-10-10 05-47-56

## Optional Changes
- [ optional ] It will be nice to ignore all files that are not relevant to run this project(eg .lintrc, .stylelintrc.json, .hintrc, .eslintrc.json etc).

**Happy coding :computer:** 
**_Do not hesitate to make comments in this PR in case of doubts._**
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by wandji20 — APPROVED** [review]
## Hi @Chike1990  :raised_hands:

##### You have done exceptionally well, and have met all requirements for this project :+1:

# Status: Approved 👍 :partying_face: 
Your project has been approved, however, I have left some comments to help improve your project.

## Congratulations :+1:  :fireworks: 
## Optional
- [ optional ] Please use this screenshot to improve your project.
Screenshot from 2021-10-10 06-27-48
- [ optional ] It will be nice to ignore all files that are not relevant to run this project(eg .lintrc, .stylelintrc.json, .eslintrc.json, etc).

_Go ahead and merge this branch. Also, don't forget to submit the project completion form.
Good luck with your next project :mechanical_arm:  :slightly_smiling_face:_ 

 If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
